src/main/java/com/jorgepatrick/NumberConverter.java

```diff
@@ -72,25 +82,41 @@ public int parseRomanToArabic(final String romanNumber) {
 
         romanNumberValidator.validateRomanNumber(romanNumber);
 
-        int unit = 0;
+        int arabicNumber = 0;
 
         for (int currentDigit = 0; currentDigit < romanNumber.length(); currentDigit++) {
-            if (romanNumber.toUpperCase().charAt(currentDigit) == I.value().charAt(0)) {
-                int nextDigit = currentDigit + 1;
-                if (nextDigit == romanNumber.length() ||
-                    romanNumber.toUpperCase().charAt(nextDigit) == I.value().charAt(0)) {
-                    unit += 1;
-                } else {
-                    unit -= 1;
-                }
+
+            if (Arrays.stream(RomanOnes).toList().contains(romanNumber.substring(currentDigit, currentDigit + 1))) {
+                int digit = Arrays.stream(RomanOnes).toList().indexOf(romanNumber.substring(currentDigit, currentDigit + 1));
```

Comment:
I think these two lines above look weird, too many dots for my taste, I was only trying to see if an array of chars contains an especific char and/or have in whitch position that char is. Is there a better way to code it? @chicofaccin

Reply:
You can use IntelliJ to help you to refactor the code and extract part of the code to very tiny methods to improve the readability. Another thing, instead of using Lists and searching all the time, you may use a Map. It's a different data structure where you can fetch the data by the index instead of searching every time.